BedrockServer.cpp

```diff
@@ -2170,8 +2157,25 @@ unique_ptr<BedrockCommand> BedrockServer::buildCommandFromRequest(SData&& reques
         request["_source"] = ip;
     }
 
+    // Pull any serialized https requests off the requests object to apply to the command.
+    string serializedHTTPSRequests = request["httpsRequests"];
+    string serializedData = request["serializedData"];
+    request.erase("httpsRequests");
+    request.erase("serializedData");
+
     // Create a command.
     unique_ptr<BedrockCommand> command = getCommandFromPlugins(move(request));
+
+    // Apply HTTPS requests. If we had any, pretend we peeked this command in our current (likely LEADING) state.
+    if (serializedHTTPSRequests.size()) {
+        command->deserializeHTTPSRequests(serializedHTTPSRequests);
+        command->lastPeekedOrProcessedInState = _syncNode->getState();
```

Comment:
I see elsewhere we set this variable we use `_server->getState()` - why do we use `_syncNode` here?

Reply:
`_syncNode->getState();` is the underlying data for `_server->getState()` (see here):
https://github.com/Expensify/Bedrock/blob/66e0c3d16b20af54da8342440aaac56128849e3d/BedrockServer.cpp#L282-L283

and in `BedrockServer::getState`:
https://github.com/Expensify/Bedrock/blob/66e0c3d16b20af54da8342440aaac56128849e3d/BedrockServer.cpp#L2377-L2379

So the syncNode state is potentially "fresher". But `_syncNode` is private and not available from outside of `BedrockServer`. I think they will always be the same here anyway, so we could use either.